### PR TITLE
chore(examples): integrate in CI flow with lint and build

### PR DESCRIPTION
**Pull request by flochaz**
## Description of your changes

(built on top of #439)

Examples were not linted or built leading to missed bugs and compilation errors.

This PR homoganize the structure of cdk examples as well as make them lint and built the same way as modules.

It also add a job in e2e test workflow to test compilation of future published packages and avoid this kind of needed fix https://github.com/awslabs/aws-lambda-powertools-typescript/pull/470 

### How to verify this change

```
npm run lerna-lint
npm run lerna-test
```

### Related issues, RFCs

To catch #435 
Built on top of #439

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dreamorosi — APPROVED** [review]
Only a couple of minor suggestions on the `CONTRIBUTING.md` file, otherwise lgtm.

**Review by dreamorosi — DISMISSED** [review]

File: examples/cdk/package.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
     "cdk-app": "bin/cdk-app.js"
   },
   "scripts": {
     "build": "tsc --skipLibCheck",
     "watch": "tsc -w",
-    "test": "jest",
+    "test": "npm run build && jest",
     "cdk": "cdk"
   },
   "devDependencies": {
```
Comment (dreamorosi) [inline examples/cdk/package.json:13]: Let's bump `@aws-lambda-powertools/commons` & others to latest version `^0.3.1`

File: examples/cdk/package.json [hunk 2]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "cdk-app",
-  "version": "0.1.0",
+  "version": "0.2.0",
```
Comment (dreamorosi) [inline examples/cdk/package.json:3]: Is this expected to follow same version of the rest of utils?

File: lerna.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "packages": [
-    "packages/*"
+    "packages/*",
+    "examples/cdk"
```
Comment (dreamorosi) [inline lerna.json:4]: Since we are importing the published versions why is this needed?

Comment (flochaz) [reply lerna.json:4]: The idea is to take advantage of the same quality checker we have in modules: husky, lint and test with lerna locally and in GH actions etc.

File: examples/cdk/package.json (hunk 2 above)
Comment (flochaz) [reply examples/cdk/package.json]: done in https://github.com/awslabs/aws-lambda-powertools-typescript/pull/446/commits/2852495933b01ababd5092c057e2a548711b7821

File: examples/cdk/package.json (hunk 1 above)
Comment (flochaz) [reply examples/cdk/package.json:15]: done in https://github.com/awslabs/aws-lambda-powertools-typescript/pull/446/commits/2852495933b01ababd5092c057e2a548711b7821

**Review by dreamorosi — APPROVED** [review]

File: lerna.json (hunk 3 above)
Comment (dreamorosi) [reply lerna.json:4]: oh good point, thanks for explaining!

**Review by dreamorosi — APPROVED** [review]
LGTM :tada:

**Review by dreamorosi — APPROVED** [review]

File: examples/cdk/package.json [hunk 4]
```diff
@@ -34,7 +34,8 @@
     "@middy/validator": "^2.5.3",
     "@types/aws-lambda": "^8.10.86",
     "aws-cdk-lib": "2.0.0",
+    "aws-sdk": "^2.1058.0",
```
Comment (saragerion) [inline examples/cdk/package.json:38]: Any specific reason why we are not using the AWS sdk v3? https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/index.html

Comment (flochaz) [reply examples/cdk/package.json:38]: we are using both in the examples to show how to capture both with tracer module

File: .github/workflows/run-e2e-tests.yml [hunk 5]
```diff
@@ -28,3 +28,9 @@ jobs:
         aws-region: eu-west-1
     - name: Run integration tests
       run: npm run lerna-test:e2e
+    - name: Test packaging
+      run: | 
+        npm run lerna-package
+        cd examples/cdk
+        npm install ../../packages/**/dist/aws-lambda-powertools-*
```
Comment (ijemmy) [inline .github/workflows/run-e2e-tests.yml:35]: My understanding is that this install ALL packages from local and ALSO modify `package.json`? Why use the local version instead of the last version?

I tried running this and it only updates the `commons` package.  `logger` in `node_modules` is still 0.2.0. 

image

image

If this isn't intentional, we can just install each module one by one.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -147,6 +147,30 @@ You can run the end-to-end tests automatically on your forked project by followi
...
+
+As part of the repo you will find an examples folder at the root. This folder contains examples (written with CDK for now) of deployable AWS Lambda functions using Powertools.
+
+To test your updates with these examples you just have to:
+
+1. Build your local version of *aws-lambda-powertools-typescript* npm packages with `npm run lerna-package`
+1. Update their references in examples
+    ```
+    cd examples/cdk
+    npm install ../../packages/**/dist/aws-lambda-powertools-*
```
Comment (ijemmy) [inline CONTRIBUTING.md:162]: Same as above.

**Review by ijemmy — CHANGES_REQUESTED** [review]
Could you double check the `npm install ..` . I think it only installs `commons` package.

File: CONTRIBUTING.md (hunk 6 above)
Comment (flochaz) [reply CONTRIBUTING.md:162]: hmm weird . it worked on mine. will double check

Comment (ijemmy) [reply CONTRIBUTING.md:162]: My bad, I didn't rerun npm run lerna-ci and a package was missing.

**Review by ijemmy — APPROVED** [review]
